### DISCUSSION TITLE
Building using bazel error:  Unrecognized option: --useDataBindingAndroidX

### DISCUSSION THREAD

**Discussion by JishnuGoyal** [oppia/oppia-android#4141] [Q&A (Installation)] [answered]
When I used `bazel build //:oppia` , it threw this error :
`Error parsing command line: Unrecognized option: --useDataBindingAndroidX`

![image](https://user-images.githubusercontent.com/64526117/151613921-431c4fd7-9e8d-4d91-a035-d97a628e7047.png)

**Comment by JishnuGoyal** [comment, marked as answer]
I finally found a solution to this!

while executing these set of commands, 

`echo build --override_repository=android_tools="$(cd "$(dirname "$HOME/opensource/oppia-bazel-tools")"; pwd)/$(basename "$HOME/opensource/oppia-bazel-tools")" > ~/.bazelrc` and
`echo build --android_databinding_use_androidx >> ~/.bazelrc`

replace $HOME with the location where your `HOME` environment variable points. That is, your root location. For ex, for me it was `/home/jishnu`.
If you are not sure what is your root location, 
run this in terminal:
`printenv HOME`
and replace the output with $HOME in the above commands everywhere

Hope this helps!

**Comment by BenHenning** [comment]
@JishnuGoyal did ``$HOME`` not expand to its value originally? That's rather unexpected if so. Which version of Linux are you running?

This might be something worth updating in https://github.com/oppia/oppia-bazel-tools#readme, but I'm on the fence since we're also waiting for Bazel 5.x (which will replace needing to use our separate tools repo altogether).

**Reply by JishnuGoyal** [reply]
I'm using Ubuntu 20.04 LTS version, and $HOME didn't expand itself automatically on my system

**Reply by BenHenning** [reply]
Thanks for the specifics, that's really surprising actually.
